src/v/kafka/server/quota_manager.h

```diff
@@ -80,16 +81,11 @@ class quota_manager : public ss::peering_sharded_service<quota_manager> {
     // tp_rate: throughput tracking
     // pm_rate: partition mutation quota tracking - only on home shard
     struct client_quota {
-        // TODO: last_seen and delay are not atomic at the moment even though
-        // they are read and written from multiple cores (= UB). For last_seen,
-        // we could either make it atomic or somehow check if the bucket is full
-        // (i.e. "10 window's worth of time is gone"). For delay, we could
-        // either make it atomic, or change the behaviour to be similar to what
-        // we do for fetch (delay is only used for produced) where we always
-        // enfore the delay, but we first decide on the delay before recording
-        // the incoming request and only then grab tokens for the incoming
-        // request.
-        clock::time_point last_seen;
+        // TODO: delay is not atomic at the moment even though
+        // it is read and written from multiple cores. For delay, we want to
+        // change the behaviour such that we track in connection_ctx whether to
+        // enforce the delay or not.
+        ss::lw_shared_ptr<ssx::sharded_int<uint64_t>> last_seen_ms;
```

Comment:
I think it would be best to keep this as a `clock::time_point`.

Reply:
Makes sense, it's now a `clock::time_point`.